src/pipe.rs

```diff
@@ -138,10 +168,17 @@ pub(crate) fn wake(pipe: RawFd, method: WakeMethod) {
         //   if it is closed. The second is EAGAIN, in which case the pipe is full ‒ there were
         //   many signals, but the reader didn't have time to read the data yet. It'll still get
         //   woken up, so not fitting another letter in it is fine.
-        let data = b"X" as *const _ as *const _;
+
+        let (data, amount) = if let Some(buffer) = buffer {
+            (buffer as *const _ as *const _, buffer.len())
+        }
+        else {
+            (b"X" as *const _ as *const _, 1)
+        };
+
         match method {
-            WakeMethod::Write => libc::write(pipe, data, 1),
-            WakeMethod::Send => libc::send(pipe, data, 1, libc::MSG_DONTWAIT),
+            WakeMethod::Write => libc::write(pipe, data, amount),
+            WakeMethod::Send => libc::send(pipe, data, amount, libc::MSG_DONTWAIT),
```

Comment:
All the Linux documentation I can find indicates that `send` and `write` are atomic up to `PIPE_BUF` bytes.  I'm curious to know why you believe that's not true.

Reply:
Because, while the module is called `pipe`, the thing that you can pass in doesn't have to be a pipe. It usually even is *not* a pipe, because rust stdandard library doesn't offer those, so most things here are actually stream unix domain sockets. Now, if the posix documentation (note the aim is not linux only) guarantees some minimal write size even in such cases, I'm fine being proven wrong :-). I usually expect less guarantees from APIs than what the sometimes support, to be on the safe side.

In such case, I'd want to see (in the final version) some kind of argument why the usage is correct and where in the documentation it is guaranteed.

Also, I think this particular function (even though exposed only internally) doesn't check the size.